Directory.Build.targets

```diff
@@ -105,4 +105,66 @@
+    moving properties based on versioning into a target.
+  -->
+  <Target Name="GetProductVersions">
+    <PropertyGroup>
+      <IncludePreReleaseLabelInPackageVersion Condition="'$(DotNetFinalVersionKind)' != 'release'">true</IncludePreReleaseLabelInPackageVersion>
+      <IncludePreReleaseLabelInPackageVersion Condition="'$(SuppressFinalPackageVersion)' == 'true'">true</IncludePreReleaseLabelInPackageVersion>
+      <IncludePreReleaseLabelInPackageVersion Condition="'$(IsShipping)' != 'true'">true</IncludePreReleaseLabelInPackageVersion>
+
+      <IncludeBuildNumberInPackageVersion Condition="'$(StabilizePackageVersion)' != 'true'">true</IncludeBuildNumberInPackageVersion>
+      <IncludeBuildNumberInPackageVersion Condition="'$(SuppressFinalPackageVersion)' == 'true'">true</IncludeBuildNumberInPackageVersion>
+      <IncludeBuildNumberInPackageVersion Condition="'$(IsShipping)' != 'true'">true</IncludeBuildNumberInPackageVersion>
+
+      <ProductVersionSuffix Condition="'$(IncludePreReleaseLabelInPackageVersion)' == 'true'">-$(VersionSuffix)</ProductVersionSuffix>
+      <ProductBandVersion Condition="'$(ProductBandVersion)' == ''">$(MajorVersion).$(MinorVersion)</ProductBandVersion>
+      <ProductionVersion Condition="'$(ProductionVersion)' == ''">$(ProductBandVersion).$(PatchVersion)</ProductionVersion>
+      <ProductVersion>$(ProductionVersion)$(ProductVersionSuffix)</ProductVersion>
+
+      <SharedFrameworkNugetVersion>$(ProductVersion)</SharedFrameworkNugetVersion>
+      <NuGetVersion>$(SharedFrameworkNugetVersion)</NuGetVersion>
+      <InstallersRelativePath>Runtime/$(SharedFrameworkNugetVersion)/</InstallersRelativePath>
+
+      <!--
+        By default, we are always building the nuget packages for HostPolicy, HostFXR and
+        Dotnet/AppHost. Thus, the "UseShipped*" properties (below) are always set to false.
+        However, there are scenarios when some of these components will not change (e.g. during
+        servicing, we may only change HostPolicy but not HostFXR and Dotnet/AppHost). In such cases,
+        set the appropriate "UseShipped*" property below to true and the corresponding "*Version"
+        property to the desired version so that we use the last shipped version of the package.
+      -->
+
+      <!-- The host/apphost package versions are only updated whenever there is a change in the components -->
+      <UseShippedHostPackage>false</UseShippedHostPackage>
+      <HostVersion Condition="'$(UseShippedHostPackage)' != 'true'">$(ProductVersion)</HostVersion>
+      <HostVersion Condition="'$(UseShippedHostPackage)' == 'true'">2.0.0</HostVersion>
+
+      <!-- The AppHostVersion is used for all hosts that aren't dotnet.exe -->
+      <UseShippedAppHostPackage>false</UseShippedAppHostPackage>
+      <AppHostVersion Condition="'$(UseShippedAppHostPackage)' != 'true'">$(ProductVersion)</AppHostVersion>
+      <AppHostVersion Condition="'$(UseShippedAppHostPackage)' == 'true'">2.0.0</AppHostVersion>
```

Comment:
@ViktorHofer, as discussed in #48341, I have kept this infra for now. IMO, since this infra is never used, we could still delete these hardcoded values (which are outdated `2.0.0`) based on `UseShippedAppHostPackage` conditions. i.e. delete https://github.com/dotnet/runtime/blob/98a9853/Directory.Build.targets#L131-L167 and document somewhere in markdown that the special case usage is:
`build host+packs -p:AppHostVersion=6.0.3-some.special.hotfix`
(instead of `build host+packs -p:UseShippedHostPackage=True`).

Reply:
Yes, please delete all these versions